docs/core/examples/courier.py

```diff
@@ -92,7 +92,8 @@ def filterMessage(self):
         A trivial example is included.
         """
         try:
-            m = Message(open(self.messageFilename))
+            emailParser = email.parser.Parser()
+            emailParser.parse(open(self.messageFilename))
```

Comment:
This appears to leak the file;  [`email.parser.Parser.parse` completely consumes the file's contents but doesn't close it.](https://github.com/python/cpython/blob/master/Lib/email/parser.py#L42-L58).  If my reading of that is correct, this should be wrapped in an with statement:

````python
with open(self.messageFilename) as f:
    emailParser.parse(f)
````

I realize this isn't part of a test or anything other than a stubbed-out example, but it'd be nice if our examples encouraged good resource management.